server/src/main/java/org/elasticsearch/action/synonyms/GetSynonymsAction.java

```diff
@@ -0,0 +1,165 @@
+            Objects.requireNonNull(SynonymsSetId, "Synonym set ID cannot be null");
+            this.synonymsSetId = SynonymsSetId;
+            this.from = from;
+            this.size = size;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            validationException = validatePageParam("from", from, validationException);
+            validationException = validatePageParam("size", size, validationException);
+
+            return validationException;
+        }
+
+        private ActionRequestValidationException validatePageParam(
+            String paramName,
+            int value,
+            ActionRequestValidationException validationException
+        ) {
+            if (value < 0) {
+                validationException = addValidationError("[" + paramName + "] must be a positive integer", validationException);
+            } else if (value > MAX_SEARCH_RESULTS) {
+                validationException = addValidationError(
+                    "[" + paramName + "] must not be greater than " + MAX_SEARCH_RESULTS,
+                    validationException
+                );
+            }
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(synonymsSetId);
+            out.writeInt(from);
+            out.writeInt(size);
+        }
+
+        public String synonymsSetId() {
+            return synonymsSetId;
+        }
+
+        public int from() {
+            return from;
+        }
+
+        public int size() {
+            return size;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request request = (Request) o;
+            return from == request.from && size == request.size && Objects.equals(synonymsSetId, request.synonymsSetId);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(synonymsSetId, from, size);
+        }
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final SynonymsManagementAPIService.SynonymsSetResult synonymsSetResults;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.synonymsSetResults = new SynonymsManagementAPIService.SynonymsSetResult(
+                in.readLong(),
+                in.readArray(SynonymRule::new, SynonymRule[]::new)
+            );
+        }
+
+        public Response(SynonymsManagementAPIService.SynonymsSetResult synonymsSetResult) {
+            super();
+            Objects.requireNonNull(synonymsSetResult, "Synonyms set result must not be null");
```

Comment:
I think we can remove this line, as `synonymsSetResult` can never be null

Reply:
I feel safer being defensive and avoid potential NPEs down the line on object constructor, even if that can't happen with the current code path. It doesn't hurt, shows intention, and will fail fast if future for new code paths.

If you think this is better to be left out, or this is a pattern in the codebase, I'm happy to remove these 👍

Reply:
It is a tiny check, so I guess it doesn't hurt to have it for future changes.